openmdao/core/explicitcomponent.py

```diff
@@ -28,8 +35,26 @@ def __init__(self, **kwargs):
         """
         super(ExplicitComponent, self).__init__(**kwargs)
 
-        if overrides_method('compute_jacvec_product', self, ExplicitComponent):
-            self.matrix_free = True
+        self._inst_functs = {name: getattr(self, name, None) for name in _inst_functs}
+
+    def _configure(self):
+        """
+        Configure this system to assign children settings.
+
+        Also tag component if it provides a guess_nonlinear.
```

Comment:
we're not tagging things with guess_nonlinear in explicit comps are we? (probably not your changed, but since I noticed it)

Reply:
no we're not, but I fixed the comment.